doc/index.rst

```diff
@@ -366,6 +372,45 @@ Special typing primitives
    raises a :py:exc:`DeprecationWarning` when this syntax is used in Python 3.12
    or lower and fails with a :py:exc:`TypeError` in Python 3.13 and higher.
 
+   ``typing_extensions`` supports the experimental additions to ``TypedDict``
+   proposed by :pep:`705`. These are implemented in the following attributes::
+
+   .. attribute:: __readonly_keys__
+
+      A :py:class:`frozenset` containing the names of all read-only keys. Keys
+      are read-only if they are declared in a ``TypedDict`` with the
+      ``readonly=True`` argument, or if they carry the :data:`ReadOnly` qualifier.
+
+      .. versionadded:: 4.9.0
+
+   .. attribute:: __mutable_keys__
+
+      A :py:class:`frozenset` containing the names of all mutable keys. Keys
+      are mutable if they are declared in a ``TypedDict`` with the
+      ``readonly=False`` argument, or if they do not carry the :data:`ReadOnly`
+      qualifier.
+
+      .. versionadded:: 4.9.0
+
+   .. attribute:: __readonly__
+
+      Boolean indicating whether the current class was declared with
+      the ``readonly=True`` argument. If this is true, all keys are read-only.
+
+      .. versionadded:: 4.9.0
+
+   .. attribute:: __other_keys__
+
+      Boolean indicating the value of the ``other_keys=`` argument for
+      the current class, which indicates whether the ``TypedDict`` accepts
+      keys other than those explicitly declared in the class body.
+
+      Note that if a base class has ``other_keys=False``,
+      but the current class does not, semantically the class will not accept
+      other keys, but this attribute will still be ``False``.
```

Comment:
I don't understand this sentence. So,

```python
class Super(TypedDict, other_keys=False): ...
class Sub(TypedDict, other_keys=True): ...
```

Does that mean that `Sub` semantically(?) doesn't allow other keys (but practically it does?) and `Sub.__other_keys__` is `False`?

Reply:
Your understanding is correct, but I agree this would be confusing. I'll leave the decision to @alicederyn as the PEP author, but I'd recommend one of these two:

- Make it an error for an `other_keys=True` TypedDict to inherit from an `other_keys=False` TypedDict. This would mean that any subclass of an `other_keys=False` TypedDict must also set `other_keys=False`. This is my preference as it's most explicit.
- Automatically set `other_keys=False` if any base class has `other_keys=False`.

Reply:
Good point. I've also gone for explicit, and updated PEP-705.

Reply:
No longer relevant as `other_keys` was removed